File: CBTC/Controller/Mutations.js
let CATCH_REQUEST = (username, species) =>
`mutation{
     catchCreature(username : ${"\""+username+"\""}, species : ${"\""+species+"\""})
}`
let COMPLETE_USER_CREATION  = (username, id, avatar) =>
`mutation{ 
    finalizeUserCreation (username : ${"\""+username+"\""}, id : ${id}, avatar : ${"\""+avatar+"\""})
}`
let SELL_ONE_CREATURE = (username, species, creatureName) => 
`mutation{ 
    sellOneCreature(username : ${"\""+username+"\""}, species : ${"\""+species+"\""}, creatureName : ${"\""+creatureName+"\""})
}`

let SELL_ALL_CREATURES = (username) => 
`mutation{ 
    sellAllCreatures(username : ${"\""+username+"\""})
}`
let ACKNOWLEDGE_TRANSACTION = (username, business, quantity, marketPrice, totalBells) => 
`mutation{ 
    acknowledgeTransaction(username : ${"\""+username+"\""}, business: ${"\""+business+"\""}, quantity: ${quantity}, marketPrice: ${marketPrice}, totalBells: ${marketPrice})
}`

module.exports.CATCH_REQUEST = CATCH_REQUEST
module.exports.COMPLETE_USER_CREATION  = COMPLETE_USER_CREATION 
module.exports.SELL_ONE_CREATURE  = SELL_ONE_CREATURE 
module.exports.SELL_ALL_CREATURES = SELL_ALL_CREATURES
module.exports.ACKNOWLEDGE_TRANSACTION = ACKNOWLEDGE_TRANSACTION

File: cbas/src/main/scala/GraphQL/Mutations.scala
package GraphQL
import Model.MovementRecord_.toggleArgs
import Model.TurnipTransaction_._
import Model.User_._
import Service._
import zio.{IO, UIO}

object Mutations {

	case class dummyArgs(dummy : Boolean)
	case class Mutations(
	    //User
	    catchCreature :           catchCreatureArgs => IO[NotFound, String],
	    finalizeUserCreation :     finalizeUserArgs => IO[NotFound, String],
	    sellOneCreature :         sellCreatureArgs => UIO[Int],
//	    sellOneCreatureByName:    sellCreatureByNameArgs => IO[NotFound, Int],
	    sellAllCreatures :        usernameArgs => UIO[Int],
	    acknowledgeTransaction:   authorizedTransactionArgs => IO[NotFound, String],
	    populate:                 dummyArgs => UIO[String],
	    toggleMarket:             toggleArgs => UIO[String],

	)
	val cbs : CrossingBotService = new CBS()

	val allMutations = Mutations(
		args => cbs.catchCreature(args.username, args.species),
		args => cbs.finalizeUserCreation(args.username, args.id, args.avatar),
		args => cbs.sellOneCreature(args.username, args.species, args.creatureName),
//		args => cbs.sellOneCreatureByName(args.username, args.creatureName),
		args => cbs.sellAllCreatures(args.username),
		args => cbs.acknowledgeTransaction(args.username, args.business, args.quantity, args.marketPrice, args.totalBells),
		args => cbs.populate(args.dummy),
		args => cbs.toggleMarket(args.running),
	)
}
